{"cells":[{"metadata":{"trusted":true},"cell_type":"code","source":"file = open(\"dataset.txt\", \"rt\")\ndata = file.read()","execution_count":1,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"data","execution_count":2,"outputs":[{"output_type":"execute_result","execution_count":2,"data":{"text/plain":"'Return of the Coronavirus: 2019-nCoV\\n\\nLisa E. Gralinski and Vineet D. Menachery\\n\\nAbstract: The emergence of a novel coronavirus (2019-nCoV) has awakened the echoes of SARS-CoV\\nfrom nearly two decades ago. Yet, with technological advances and important lessons gained from\\nprevious outbreaks, perhaps the world is better equipped to deal with the most recent emergent\\ngroup 2B coronavirus.\\n\\nKeywords: 2019-nCoV; novel CoV; Wuhan; Wuhan pneumonia; coronavirus; emerging viruses;\\nSARS-CoV; MERS-CoV\\n\\n1. Emergence\\n\\nThe third zoonotic human coronavirus (CoV) of the century emerged in December 2019, with\\na cluster of patients with connections to Huanan South China Seafood Market in Wuhan, Hubei\\nProvince, China. Similar to severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle\\nEast respiratory syndrome coronavirus (MERS-CoV) infections, patients exhibited symptoms of viral\\npneumonia including fever, diﬃculty breathing, and bilateral lung inﬁltration in the most severe\\ncases [1]. News reports of patients with an unknown pneumonia were ﬁrst identiﬁed on 31st December\\nwith the Wuhan Municipal Health Commission saying they were monitoring the situation closely\\n(Figure 1). On 1st January 2020, the seafood market was closed and decontaminated while countries\\nwith travel links to Wuhan went on high alert for potential travelers with unexplained respiratory\\ndisease. After extensive speculation about a causative agent, the Chinese Center for Disease Control\\nand Prevention (CDC) conﬁrmed a report by the Wall Street Journal and announced identiﬁcation\\nof a novel CoV on 9th January [2]. The novel CoV (2019-nCoV) was isolated from a single patient\\nand subsequently veriﬁed in 16 additional patients [3]. While not yet conﬁrmed to induce the viral\\npneumonia, 2019-nCoV was quickly predicted as the likely causative agent.\\n\\nThe ﬁrst sequence of 2019-nCoV was posted online one day after its conﬁrmation on behalf of\\nDr. Yong-Zhen Zhang and scientists at Fudan University, Shanghai [4]. Subsequently, ﬁve additional\\n2019-nCoV sequences were deposited on the GSAID database on 11th January from institutes across\\nChina (Chinese CDC, Wuhan Institute of Virology and Chinese Academy of Medical Sciences &\\nPeking Union Medical College) and allowed researchers around the world to begin analyzing the new\\nCoV [5]. By 17th January, there were 62 conﬁrmed cases in China and importantly, three exported\\ncases of infected travelers who were diagnosed in Thailand (2) and Japan (1) [6]. The sequences of\\nthese exported cases and several additional 2019-nCoV isolated in China have also been deposited\\non the GSAID database [5]. Diagnostic tests have subsequently been developed and some are being\\nused on suspect cases identiﬁed in other locations including Vietnam, Singapore, and Hong Kong [7].\\nTo date there have been twenty-six fatalities associated with 2019-nCoV infection, many of these cases\\nhad signiﬁcant co-morbidities and were older in age (>50). A range of disease has been observed\\nhighlighted by fever, dry cough, shortness of breath, and leukopenia; patients have included mild cases\\nneeding supportive care to severe cases requiring extracorporeal membrane oxygenation; however,\\ncompared to SARS-CoV (10% mortality) and MERS-CoV (35% mortality), the 2019-nCoV appears to be\\nless virulent at this point with the exception of the elderly and those with underlying health conditions.\\nInitial monitoring of case close contacts had not revealed any further 2019-nCoV cases. However,\\nmodeling analysis based on oﬃcial case numbers and international spread suggested that there may\\nbe cases going undetected [8]. On 19th January, these fears were seemingly conﬁrmed as an additional\\n136 cases were added from further surveys raising the total in Wuhan to 198 infected patients [9].\\nAmong the 198 total cases in Wuhan, 170 remained in hospitals, 126 mostly with mild symptoms,\\n35 in serious condition, and 9 in critical condition. The expanded numbers and extended range of\\nonset dates (12 December 2019–18 January 2020) suggested likely human to human transmission or\\nongoing transmission from a market or other primary sources. On 20th January, the outbreak was\\nfurther expanded to other parts of China (Beijing, Shanghai, & Shenzhen) as well as another exported\\ncases to South Korea. As of January 24, the total case number has expanded to at least 870 total cases\\nwith 26 deaths across 25 provinces in China and 19 exported cases in 10 countries [10]. Public health\\nauthorities have quarantined travel from Wuhan to limit the spread of the virus and reports indicate\\nother Chinese cities have also been isolated [11]. With the heavy travel season for lunar New Year\\nunderway in Asia, major concerns exist for the 2019-nCoV outbreak to continue and spread.\\n\\n2. Origins of 2019-nCoV\\n\\nThe source of the 2019-nCoV is still unknown, although the initial cases have been associated with\\nthe Huanan South China Seafood Market. While many of the early patients worked in or visited the\\nmarket, none of the exported cases had contact with the market, suggesting either human to human\\ntransmission or a more widespread animal source [6]. In addition to seafood, it is reported on social\\nmedia that snakes, birds and other small mammals including marmots and bats were sold at the\\nHuanan South China Seafood Market. The WHO reported that environmental samples taken from the\\nmarketplace have come back positive for the novel coronavirus, but no speciﬁc animal association\\nhas been identiﬁed [6]. An initial report suggested that snakes might be the possible source based\\non codon usage [12], but the assertion has been disputed by others [13]. Researchers are currently\\nworking to identify the source of 2019-nCoV including possible intermediate animal vectors.\\n\\nA zoonotic reservoir harkens back to the emergence of both SARS- and MERS-CoV. SARS-CoV,\\nthe ﬁrst highly pathogenic human CoV, emerged in 2002 with transmission from animals to humans\\noccurring in wet markets. Surveillance eﬀorts found SARS-CoV viral RNA in both palm civets and\\nraccoon dogs sold in these wet markets [14]; however, SARS-CoV was not found in the wild, suggesting\\nthat those species served as intermediary reservoir as the virus adapted to more eﬃciently infect\\nhumans. Further surveillance eﬀorts identiﬁed highly related CoVs in bat species [15]. More recent\\nwork has demonstrated that several bat CoVs are capable of infecting human cells without a need for\\nintermediate adaptation [16,17]. Additionally, human serology data shows recognition of bat CoV\\nproteins and indicates that low-level zoonotic transmission of SARS-like bat coronaviruses occurs\\noutside of recognized outbreaks [18]. MERS-CoV is also a zoonotic virus with possible origins in\\nbats [19,20], although camels are endemically infected and camel contact is frequently reported during\\nprimary MERS-CoV cases [21]. For SARS-CoV, strict quarantine and the culling of live markets in SE\\nAsia played a major role in ending the outbreak. With the cultural importance of camels, a similar\\napproach for MERS-CoV was not an option and periodic outbreaks continue in the Middle East. These\\nlessons from SARS and MERS highlight the importance of rapidly ﬁnding the source for 2019-nCoV in\\norder to stem the ongoing outbreak.\\n\\n3. Susceptible Populations\\n\\nWith limited patient data, it is diﬃcult to make robust declarations about populations that\\nmay be most susceptible to 2019-nCoV. However, disease severity following SARS- and MERS-CoV\\ncorresponded strongly to underlying host conditions including age, biological sex, and overall\\nhealth [22]. Early patient reports from 2019-nCoV ﬁnd similar trends. Severe illness with 2019-nCoV has\\nbeen associated with elderly patients (>60 years old), including twenty-six lethal cases. These ﬁndings\\ncorrespond to increased severity and death in people over the age of 50 following both SARS and\\nMERS-CoV infection [23,24]. Similarly, the underlying health of the patient likely plays a critical\\nrole in overall susceptibility. For the 2019-nCoV, limited comorbidity data is available; however,\\nthe twenty-six patients that have succumbed to the novel CoV had signiﬁcant health conditions\\nincluding hypertension, diabetes, heart and/or kidney function issues that may have made them more\\nsusceptible. For the MERS-CoV outbreak, smoking, hypertension, diabetes, cardiovascular disease,\\nand/or other chronic illnesses have been present in the majority of deaths and correspond to ﬁndings in\\nanimal models [25]. The results indicate vigilance is necessary for these vulnerable patients following\\n2019-nCoV infection.\\n\\n4. Insights from the 2019-nCoV Sequence\\n\\nThe rapid sequencing of the nearly 30,000 nucleotide 2019-nCoV genome by Dr. Zhang’s group at\\nFudan University and several other groups in China illustrate the dedication and increased capacity\\nof the scientiﬁc infrastructure in China [4,5]. For SARS-CoV, the causative agent was unknown for\\nmonths and subsequently took over four weeks until a full genome was released [26]. Similarly,\\nMERS-CoV was only identiﬁed after several months of testing and a full-length genome available\\nabout a month later [27]. In contrast, time from the ﬁrst date of patient onset (12 December 2019) to\\nthe report of several 2019-nCoV full-length genomes took less than one month. Combined with the\\nimmense pressure of an ongoing outbreak with an unknown agent, the eﬀort of these scientists should\\nbe considered nothing less than remarkable.\\n\\nBuilding from the sequence, the nucleotide alignment quickly distinguished the novel virus as a\\ngroup 2B CoV, distinct from the SARS-CoV strains [4,5]. Examining the whole genome, 2019-nCoV\\nmaintains ~80% nucleotide identity to the original SARS epidemic viruses. Its closest whole genome\\nrelatives are two bat SARS-like CoVs (ZC45 and ZXC21) that shared ~89% sequence identity with\\n2019-nCoV; these CoV sequences were deposited in early 2018 from Zhejiang province in R. sinicus\\nbats in China. Comparing across the deposited 2019-nCoV strains ﬁnds > 99.5% conservation; the lack\\nof diversity suggests a common lineage and source with emergence not likely having occurred that\\nlong ago [28,29]. A recent report has subsequently identiﬁed a bat CoV sequence, RaTG3, with 92%\\nsequence identity with the novel virus which argues for bat origins for the 2019-nCoV [30].\\n\\nWe next shifted analysis to the nucleocapsid (N) protein, the most abundant protein produced in\\nCoVs. Generally, the N protein is well conserved across CoV families including group 2B [31]. The N\\nprotein for 2019-nCoV is no exception with ~90% amino acid identity to the SARS-CoV N protein. While\\nless conserved than other group 2B CoVs like HKU3-CoV and SHC014-CoV, 2019-nCoV antibodies\\nagainst the N protein would likely recognize and bind the SARS-CoV N protein as well. N antibodies\\ndo not provide immunity to 2019-nCoV infection, but the cross reactivity with SARS-CoV N protein\\nwould allow a serum based assay to determine exposure to the novel CoV in asymptomatic cases. While\\nprevious studies have found serum reactivity to group 2B virus N proteins in Chinese populations [18],\\nexposure to 2019-nCoV should increase the dilution factor substantially if exposure/infection had\\noccurred. Importantly, this information may provide insights about susceptibly and potential routes of\\nspread through asymptomatic carriers.\\n\\nExamining further, we next compared the spike proteins, the critical glycoprotein responsible\\nfor virus binding and entry. Overall, the 2019-nCoV spike protein has roughly 75% amino acid\\nidentity with SARS-CoV, which is less conserved than other group 2B CoVs including HKU3-CoV [31].\\nHowever, narrowing analysis to the spike receptor binding domain (RBD) of SARS-CoV (amino acids\\n318–518), the 2019-nCoV RBD is 73% conserved relative to the epidemic RBD. This conservation\\nlevel places the 2019-nCoV RBD between HKU3-4 (62.7% conservation), a bat virus that cannot use\\nhuman ACE2, and rSHC014 (80.8%), the most divergent bat CoV spike known to use human ACE2 for\\nentry [16,32]. Importantly, the key binding residues for SARS-CoV have been identiﬁed [33]; among\\nthese fourteen residues predicted to interact directly with human ACE2, the receptor for SARS-CoV,\\neight amino acids are conserved in 2019-nCoV. Notably, several of these residues are also conserved\\nrelative to WIV1- and WIV16-CoV, two bat strains closely related to SARS-CoV and known to use\\nhuman ACE2 [17,34]. Initial structural modeling suggest that the 2019-nCoV may be able to use human\\nACE2 as a receptor, although its aﬃnity m be reduced relative to the epidemic SARS-CoV strains [35].\\nA subsequent report demonstrated that the receptor binding domain of 2019-nCoV was capable of\\nbinding ACE2 in the context of the SARS-CoV spike protein [36]. In addition, another rapid report\\nlinks demonstrates 2019-nCoV uses ACE2 receptors from human, bat, civets, and swine [30]. Together,\\nthe modeling, pseudotyping, and infection data provide strong evidence for human ACE2 being the\\nreceptor for 2019-nCoV.\\n\\n5. Achieving Koch Postulates\\n\\nTraditional identiﬁcation of a microbe as the causative agent of disease requires fulﬁllment of\\nKoch’s postulates, modiﬁed by Rivers for viral diseases [37]. At the present time, the 2019-nCoV\\nhas been isolated from patients, detected by speciﬁc assays in patients, and cultured in host cells\\n(one available sequence is identiﬁed as a passage isolate), starting to fulﬁll these criteria. Given the\\nrecentness of the 2019-nCoV outbreak, at this point there is no animal model available to fulﬁll the\\nremaining criteria: 1) testing the capability of 2019-nCoV to cause respiratory disease in a related\\nspecies, 2) re-isolating the virus from the experimentally infected animal and 3) detection of a speciﬁc\\nimmune response. These eﬀorts will surely be an area of intense research in the coming months both in\\nChina and in CoV research laboratories around the world.\\n\\nNotably, generating small animal models of coronavirus disease can be diﬃcult. While SARS-CoV\\nreadily infected laboratory mice, it does not cause signiﬁcant disease unless the virus is passaged to\\nadapt to the mouse host [38]. Infection of primates produces a more mild disease than that observed in\\nhumans, although fever and pulmonary inﬂammation were noted [39,40]. MERS-CoV is incapable\\nof infecting rodent cells without engineering changes in critical residues of the receptor protein,\\nDPP4 [41,42]. However, MERS-CoV does infect non-human primates [43]. As such, MERS mouse\\nmodels of disease required a great deal of time to develop and are limited in the types of manipulations\\nthat can be performed [41]. At this point, the infectious capability of the 2019-nCoV for diﬀerent species\\nand diﬀerent cell types is unknown. Early reports suggest that the virus can utilize human, bat, swine,\\nand civet ACE2 [30]; notably, the group found mouse Ace2 was not permissive for 2019-nCoV infection\\nDissemination of virus stocks and/or de novo generation of the virus through reverse genetics systems\\nwill enable this research allowing for animal testing and subsequent completion of Koch’s postulates\\nfor the new virus.\\n\\n6. Threat for Spread: Human to Human, Health Care Workers, and Super Spreaders\\n\\nWhile the Huanan seafood market in Wuhan has been associated with the majority of cases, many\\nof the recent cases do not have a direct connection [9]. This fact suggests a secondary source of infection,\\neither human to human transmission or possibly infected animals in another market in Wuhan. Both\\npossibilities represent major concerns and indicate the outbreak has the potential to expand rapidly.\\nFor human to human transmission, there was limited data in the initial set of cases; one family cluster\\nis of three men who all work in the market. Similarly, a husband and wife are among the patients, with\\nthe wife claiming no contact with the market. In these cases, direct human to human infection may\\nhave been possible; alternatively, a contaminated fomite from the market may also be responsible as\\nsurfaces all around the market were found to test positive 2019-nCoV. However, the major increase\\nin the number of cases, the lack of direct connection to the Wuhan market for many cases, and the\\ninfection of health care works all suggest human to human spread is likely [9,44]. Importantly, until the\\nsource of the virus is found, it will be diﬃcult to distinguish zoonotic versus human to human spread.\\nIn the early part of the outbreak, the absence of infection in health care workers argued for ineﬃcient\\nhuman to human spread and distinguished 2019-nCoV from both SARS-CoV and MERS-CoV. In\\nthe two prior CoV epidemics, health care settings served as a major transmission point fueling both\\noutbreaks. Based on WHO data, 1 in 10 MERS-CoV cases have been found to be health care workers;\\nthese patients generally have reduced disease and death likely due to younger age and absence of\\nexisting health conditions. The recent reports of numerous infected health care workers in Wuhan\\nindicate human to human infection can occur with 2019-nCoV and may be the product of a super\\nspreading patient [44]. However, while large swaths of healthcare workers are not getting sick as seen\\nwith SARS and MERS-CoV, it may be too early to rule out their potential exposure to the novel CoV as\\ntheir disease may be asymptomatic. While not described during the SARS-CoV outbreak, asymptomatic\\ncases ranged from 12.5% to 25% in some MERS-CoV studies [45]. A similar phenomenon may be\\noccurring with 2019-nCoV and would make stopping the outbreak even more diﬃcult to contain.\\n\\nAnother parameter to consider is the possibility of super spreading in the context of 2019-nCoV.\\nSuper spreading is the ampliﬁed transmission of a virus by individuals in a population and has been\\nsuggested by at least one news report [44]. Both SARS- and MERS-CoV outbreaks had documented\\nevidence of super spreading patients [46]. In general, both epidemic CoVs maintain a low R0, the\\nrate spread from an individual infected patient. However, roughly 10% of SARS- and MERS-CoV\\n> 10. These cases seeded a signiﬁcant\\npatients have been associated with super spreading and an R0\\nportion of the epidemic around the world. Notably, neither mutations in the viruses nor severity of\\ndisease were found to be associated with super spreading, implying that host factors contribute to the\\nphenotype [47]. For 2019-nCoV, contact tracing to date suggest limited human to human spread and a\\nlow R0. However, the recent increase in cases, both in and outside Wuhan could signal the existence of\\nsuper-spreading individuals fueling the outbreak. Alternatively, super spreading could occur from the\\nzoonotic source which has been seen in other disease outbreaks [10]. In any event, the possibility of\\nsuper spreading may continue to play a role in this ongoing 2019-nCoV outbreak.\\n\\n7. Emerging Diseases in the Age of Social Media\\n\\nNews of the 2019-nCoV came to widespread attention through the internet. Over the years,\\nwebsites like FluTrackers.com, ProMED (promedmail.org), and others have permitted the collection of\\ndisease information from around the world and facilitated dissemination to interested parties. In 2012,\\nMERS-CoV ﬁrst drew attention as a “novel coronavirus” entioned on ProMED Mail and subsequently\\nthrough conversation on twitter between science journalists, virologists, and public health experts.\\nEight years later, a more connected network quickly dissected statements from the Wuhan Municipal\\nHealth Commission and speculated about possible causes. Early during an outbreak, it can be diﬃcult\\nto distinguish between rumors with elements of truth versus baseless fear mongering. This fact can\\nbe exacerbated by language barriers and oﬀ the record sources. However, in this case, speculation of\\na novel coronavirus was fed by carefully worded statements that speciﬁcally excluding some virus\\nfamilies (inﬂuenza, adenovirus), but only excluded SARS-CoV and MERS-CoV for coronaviruses.\\nCoupled with memories of the SARS outbreak, many worried that the truth may be held back. When\\nthe agent was ﬁnally conﬁrmed as a CoV, the world acted with both worry and relief: the outbreak\\nwould not be hidden.\\n\\nWhile far from perfect, the government response to 2019-nCoV provides a stark contrast to the\\nSARS outbreak at the beginning of the century. The rapid release of 2019-nCoV sequences permitted the\\nresearch community to quickly become engaged, providing analysis and developing diagnostic tests.\\nBoth the Chinese CDC and the Wuhan Municipal Health Commission have posted regular updates\\nof conﬁrmed case numbers and patient statuses enabling public health authorities to monitor the\\nsituation in real time. Researchers from around the world have connected on social media to compare\\nupdated sequence information and highlight key unknowns about the outbreak. While not always\\nprovided in a timely manner, the ability to share news updates and data in real time with researchers\\nand public health oﬃcials around the world signals a major change in the response to outbreaks. This\\nconnectivity has facilitated awareness as well as new collaborations and a rapid response by the global\\nresearch community. While there are many unknowns with 2019-nCoV, the world is engaged and\\nprepared to battle the newest emergent virus strain. Perhaps this means the lessons from the SARS\\noutbreak have truly been learned.\\n\\nAuthor Contributions: L.E.G. and V.D.M. prepared this manuscript. All authors have read and agreed to the\\npublished version of the manuscript.\\n\\nFunding: This research received no external funding.\\n\\nConﬂicts of Interest: The authors declare no conﬂict of interest.\\n\\n© 2020 by the authors. Licensee MDPI, Basel, Switzerland. This article is an open access\\narticle distributed under the terms and conditions of the Creative Commons Attribution\\n(CC BY) license (http://creativecommons.org/licenses/by/4.0/).\\n\\n\\x0c'"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"pip install nltk","execution_count":4,"outputs":[{"output_type":"stream","text":"Collecting nltk\n  Downloading nltk-3.6.2-py3-none-any.whl (1.5 MB)\n\u001b[K     |████████████████████████████████| 1.5 MB 3.6 MB/s eta 0:00:01     |▊                               | 30 kB 3.9 MB/s eta 0:00:01\n\u001b[?25hRequirement already satisfied: joblib in /srv/conda/envs/notebook/lib/python3.6/site-packages (from nltk) (1.0.1)\nCollecting regex\n  Downloading regex-2021.7.1-cp36-cp36m-manylinux2014_x86_64.whl (722 kB)\n\u001b[K     |████████████████████████████████| 722 kB 70.6 MB/s eta 0:00:01\n\u001b[?25hCollecting click\n  Downloading click-8.0.1-py3-none-any.whl (97 kB)\n\u001b[K     |████████████████████████████████| 97 kB 9.2 MB/s  eta 0:00:01\n\u001b[?25hCollecting tqdm\n  Downloading tqdm-4.61.1-py2.py3-none-any.whl (75 kB)\n\u001b[K     |████████████████████████████████| 75 kB 7.4 MB/s  eta 0:00:01\n\u001b[?25hRequirement already satisfied: importlib-metadata in /srv/conda/envs/notebook/lib/python3.6/site-packages (from click->nltk) (3.4.0)\nRequirement already satisfied: typing-extensions>=3.6.4 in /srv/conda/envs/notebook/lib/python3.6/site-packages (from importlib-metadata->click->nltk) (3.7.4.3)\nRequirement already satisfied: zipp>=0.5 in /srv/conda/envs/notebook/lib/python3.6/site-packages (from importlib-metadata->click->nltk) (3.4.0)\nInstalling collected packages: tqdm, regex, click, nltk\nSuccessfully installed click-8.0.1 nltk-3.6.2 regex-2021.7.1 tqdm-4.61.1\nNote: you may need to restart the kernel to use updated packages.\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from nltk.corpus import stopwords","execution_count":5,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"import nltk","execution_count":7,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"nltk.download('stopwords')","execution_count":8,"outputs":[{"output_type":"stream","text":"[nltk_data] Downloading package stopwords to /home/jovyan/nltk_data...\n[nltk_data]   Unzipping corpora/stopwords.zip.\n","name":"stderr"},{"output_type":"execute_result","execution_count":8,"data":{"text/plain":"True"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from nltk.tokenize import word_tokenize","execution_count":9,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"import nltk\nnltk.download('punkt')\ntext_tokens = word_tokenize(data)","execution_count":11,"outputs":[{"output_type":"stream","text":"[nltk_data] Downloading package punkt to /home/jovyan/nltk_data...\n[nltk_data]   Unzipping tokenizers/punkt.zip.\n","name":"stderr"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"tokens_without_sw = [word for word in text_tokens if not word in stopwords.words()]\n\n#print(tokens_without_sw)\nwords1 = [word for word in tokens_without_sw if word.isalpha()]\nprint(words1[:100])","execution_count":20,"outputs":[{"output_type":"stream","text":"['Return', 'Coronavirus', 'Lisa', 'Gralinski', 'Vineet', 'Menachery', 'Abstract', 'The', 'emergence', 'novel', 'coronavirus', 'awakened', 'echoes', 'nearly', 'two', 'decades', 'ago', 'Yet', 'technological', 'advances', 'important', 'lessons', 'gained', 'previous', 'outbreaks', 'perhaps', 'world', 'better', 'equipped', 'deal', 'recent', 'emergent', 'group', 'coronavirus', 'Keywords', 'novel', 'CoV', 'Wuhan', 'Wuhan', 'pneumonia', 'coronavirus', 'emerging', 'viruses', 'Emergence', 'The', 'third', 'zoonotic', 'human', 'coronavirus', 'CoV', 'century', 'emerged', 'December', 'cluster', 'patients', 'connections', 'Huanan', 'South', 'China', 'Seafood', 'Market', 'Wuhan', 'Hubei', 'Province', 'China', 'Similar', 'severe', 'acute', 'respiratory', 'syndrome', 'coronavirus', 'Middle', 'East', 'respiratory', 'syndrome', 'coronavirus', 'infections', 'patients', 'exhibited', 'symptoms', 'viral', 'pneumonia', 'including', 'fever', 'diﬃculty', 'breathing', 'bilateral', 'lung', 'inﬁltration', 'severe', 'cases', 'News', 'reports', 'patients', 'unknown', 'pneumonia', 'ﬁrst', 'identiﬁed', 'December', 'Wuhan']\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from collections import Counter","execution_count":29,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"split_it = str(words1).split()","execution_count":30,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"Counter = Counter(split_it)\n  ","execution_count":31,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"most_occur = Counter.most_common(10)","execution_count":32,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(most_occur)","execution_count":33,"outputs":[{"output_type":"stream","text":"[(\"'human',\", 32), (\"'cases',\", 30), (\"'outbreak',\", 19), (\"'CoV',\", 18), (\"'virus',\", 18), (\"'Wuhan',\", 16), (\"'patients',\", 16), (\"'disease',\", 15), (\"'The',\", 14), (\"'health',\", 14)]\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from textblob import TextBlob\nblob = TextBlob(str(words1))\nprint(blob.noun_phrases)\n","execution_count":39,"outputs":[{"output_type":"stream","text":"[\"[ 'return\"]\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"pip install TextBlob","execution_count":35,"outputs":[{"output_type":"stream","text":"Collecting TextBlob\n  Downloading textblob-0.15.3-py2.py3-none-any.whl (636 kB)\n\u001b[K     |████████████████████████████████| 636 kB 4.8 MB/s eta 0:00:01\n\u001b[?25hRequirement already satisfied: nltk>=3.1 in /srv/conda/envs/notebook/lib/python3.6/site-packages (from TextBlob) (3.6.2)\nRequirement already satisfied: joblib in /srv/conda/envs/notebook/lib/python3.6/site-packages (from nltk>=3.1->TextBlob) (1.0.1)\nRequirement already satisfied: click in /srv/conda/envs/notebook/lib/python3.6/site-packages (from nltk>=3.1->TextBlob) (8.0.1)\nRequirement already satisfied: tqdm in /srv/conda/envs/notebook/lib/python3.6/site-packages (from nltk>=3.1->TextBlob) (4.61.1)\nRequirement already satisfied: regex in /srv/conda/envs/notebook/lib/python3.6/site-packages (from nltk>=3.1->TextBlob) (2021.7.1)\nRequirement already satisfied: importlib-metadata in /srv/conda/envs/notebook/lib/python3.6/site-packages (from click->nltk>=3.1->TextBlob) (3.4.0)\nRequirement already satisfied: typing-extensions>=3.6.4 in /srv/conda/envs/notebook/lib/python3.6/site-packages (from importlib-metadata->click->nltk>=3.1->TextBlob) (3.7.4.3)\nRequirement already satisfied: zipp>=0.5 in /srv/conda/envs/notebook/lib/python3.6/site-packages (from importlib-metadata->click->nltk>=3.1->TextBlob) (3.4.0)\nInstalling collected packages: TextBlob\nSuccessfully installed TextBlob-0.15.3\nNote: you may need to restart the kernel to use updated packages.\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"import nltk\nnltk.download('brown')\n  ","execution_count":38,"outputs":[{"output_type":"stream","text":"[nltk_data] Downloading package brown to /home/jovyan/nltk_data...\n[nltk_data]   Unzipping corpora/brown.zip.\n","name":"stderr"},{"output_type":"execute_result","execution_count":38,"data":{"text/plain":"True"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"import nltk \nfrom nltk.corpus import stopwords \nfrom nltk.tokenize import word_tokenize, sent_tokenize","execution_count":40,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"def ProperNounExtractor(text):\n    \n    \n    counter=0\n    sentences = nltk.sent_tokenize(text)\n    for sentence in sentences:\n        words = nltk.word_tokenize(sentence)\n        words = [word for word in words if word not in set(stopwords.words('english'))]\n        tagged = nltk.pos_tag(words)\n        for (word, tag) in tagged:\n            if tag == 'NNP': # If the word is a proper noun\n                counter=counter+1\n                #print(word)\n        print('The article has' +' ' +str(counter)+' '+'nouns')","execution_count":54,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"text = str(words1)\n# Calling the ProperNounExtractor function to extract all the proper nouns from the given text. \nProperNounExtractor(text)","execution_count":55,"outputs":[{"output_type":"stream","text":"The article has 921 nouns\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"import nltk\nnltk.download('averaged_perceptron_tagger')","execution_count":44,"outputs":[{"output_type":"stream","text":"[nltk_data] Downloading package averaged_perceptron_tagger to\n[nltk_data]     /home/jovyan/nltk_data...\n[nltk_data]   Unzipping taggers/averaged_perceptron_tagger.zip.\n","name":"stderr"},{"output_type":"execute_result","execution_count":44,"data":{"text/plain":"True"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3","language":"python"},"language_info":{"name":"python","version":"3.6.13","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat":4,"nbformat_minor":5}